### PR TITLE
fixes #4005 - ensure standalone Rabl.render works outside of a controller

### PR DESCRIPTION
**Pull request by domcleal**

### CODE REVIEW

File: test/unit/rabl_test.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    rendered = Rabl.render([OpenStruct.new(:name => 'foo')],
+                           'api/v2/domains/index',
+                           :format => :json,
+                           :view_path => 'app/views')
+    loaded = JSON.load(rendered)
+    assert_equal Array, loaded.class
+    assert_equal Hash, loaded[0].class
+    assert_equal 'foo', loaded[0]['name']
+  end
+end
```
Comment (dmitri-d) [inline test/unit/rabl_test.rb:25]: lgtm. I couldn't find other tests for json root inclusion logic -- probably worth testing other conditions too, as there are quite a few of them.

Comment (domcleal) [inline test/unit/rabl_test.rb:25]: Yeah, we should have some abstract RABL tests, but I think we do have coverage in specific functional API tests.

File: config/initializers/rabl_init.rb [hunk 2]
```diff
@@ -29,9 +29,9 @@ def collection_with_defaults(data, options = default_options)
     # 2) custom  - pass ?params[:root_name]=custom_name in URL
     def data_name(data_token)
       # custom object root
-      return params['root_name'] if params['root_name'].present? && !['false', false].include?(params['root_name'])
+      return params['root_name'] if self.respond_to?(:params) && params['root_name'].present? && !['false', false].include?(params['root_name'])
```
Comment (ohadlevy) [inline config/initializers/rabl_init.rb:32]: nitpick: sometime you use self.respond_to? and sometimes without

Comment (domcleal) [inline config/initializers/rabl_init.rb:32]: Any preference?  I kind of prefer `self.`

**Comment by ohadlevy** [comment]
merged as 858708c thanks!

**Pull request closed without merging**